tests/test_boundary_conditions.py

```diff
@@ -221,6 +218,162 @@ def test_fixed_constraint(seed, n_position_constraint, n_director_constraint):
     )
 
 
+@pytest.mark.parametrize("seed", [1, 10, 100])
+@pytest.mark.parametrize("num_translational_constraint", [0, 1, 2])
+@pytest.mark.parametrize("num_rotational_constraint", [0, 1, 2])
+@pytest.mark.parametrize(
+    "constraint_selector",
+    [
+        np.array([False, False, False]),
+        np.array([True, False, False]),
+        np.array([False, True, False]),
+        np.array([False, False, True]),
+        np.array([True, True, False]),
+        np.array([True, False, True]),
+        np.array([False, True, True]),
+        np.array([True, True, True]),
+    ],
+)
+def test_configurable_constraint(
+    seed,
+    num_translational_constraint,
+    num_rotational_constraint,
+    constraint_selector,
+):
+    rng = np.random.default_rng(seed)
+    test_rod = MockTestRod()
+
+    start_position_collection = rng.random((num_translational_constraint, 3))
+    start_director_collection = rng.random((num_rotational_constraint, 3, 3))
+    translational_constraint_selector = constraint_selector
+    rotational_constraint_selector = constraint_selector
+    configurable_constraint = ConfigurableConstraint(
+        *start_position_collection,
+        *start_director_collection,
+        translational_constraint_selector=translational_constraint_selector,
+        rotational_constraint_selector=rotational_constraint_selector,
+        _system=test_rod,
+    )
+    pos_indices = rng.choice(
+        test_rod.n_elem, size=num_translational_constraint, replace=False
+    )
+    dir_indices = rng.choice(
+        test_rod.n_elem, size=num_rotational_constraint, replace=False
+    )
+    configurable_constraint._constrained_position_idx = pos_indices.copy()
+    configurable_constraint._constrained_director_idx = dir_indices.copy()
+
+    test_position_collection = rng.random((3, test_rod.n_elem))
+    test_rod.position_collection = (
+        test_position_collection.copy()
+    )  # We need copy of the list not a reference to this array
+    configurable_constraint.constrain_values(test_rod, time=0)
```

Comment:
```suggestion
    general_constraint.constrain_values(test_rod, time=0)
```

Reply:
Implemented in https://github.com/GazzolaLab/PyElastica/pull/143/commits/4c06b0e850b5d5fb647394771572c35a9f0c641e